rllib/execution/train_ops.py

```diff
@@ -65,6 +65,104 @@ def train_one_step(trainer, train_batch) -> Dict:
     return info
 
 
+@ExperimentalAPI
+def multi_gpu_train_one_step(trainer, train_batch) -> Dict:
+    config = trainer.config
+    workers = trainer.workers
+    local_worker = workers.local_worker()
+    policies = local_worker.policies_to_train
+    num_sgd_iter = config.get("sgd_num_iter", 1)
+    sgd_minibatch_size = config.get("sgd_minibatch_size", 0)
+
+    # Collect actual GPU devices to use.
+    if not config.get("num_gpus", 0):
+        _fake_gpus = True
+        num_gpus = 1
+    type_ = "cpu" if _fake_gpus else "gpu"
+    devices = [
+        "/{}:{}".format(type_, 0 if _fake_gpus else i)
+        for i in range(int(math.ceil(num_gpus)))
+    ]
+
+    # Make sure total batch size is dividable by the number of devices.
+    # Batch size per tower.
+    per_device_batch_size = sgd_minibatch_size // len(devices)
+    # Total batch size.
+    batch_size = per_device_batch_size * len(devices)
+    assert batch_size % len(devices) == 0
+    assert batch_size >= len(devices), "Batch size too small!"
```

Comment:
kind of feel like we should move these asserts into the "load data" for loop below, and make sure per-policy batch sizes (the actual data we load onto each gpu) satisfy the 2 conditions here?